Document `length` for numbers and bools

alexjurkiewicz opened the pull request:
This was added in ae7a042876130c471dce28a4396abc215192eaa9 but not explicitly documented.

alexjurkiewicz committed:
Document use of length on numbers

This was added in ae7a042876130c471dce28a4396abc215192eaa9 but not explicitly documented.

wader commented:
As we're nearly listing all types, should we also add that length of boolean is an error?

alexjurkiewicz committed:
Document length behaviour on booleans

And include an example of length on numbers.

wader reviewed (approved):
Maybe update PR title?

itchyny reviewed (changes requested):
This needs to applied to the development version manual too. Also, the period is missing (sorry for nitpick).

alexjurkiewicz committed:
Add missing period

alexjurkiewicz committed:
Document `length` on numbers and bools

It was already documented in the 1.6 manual, port that over to the trunk manual.

I've updated the example to use a different number, because `6` was already used as string length which is mildly confusing.

alexjurkiewicz committed:
Update length example

The length `6` was used for two items in the example, which could be confusing.

wader reviewed (approved):
@itchyny do we care to update < 1.6 verisons too?

itchyny reviewed (approved):
LGTM, thank you.

@wader
Document improvements are good enough if been applied to the latest released & development versions. But talking of typo fixes, they keep sending patches with spell checkers even if it is in older version manuals, so I'd like to fix them all.
